### DISCUSSION TITLE
call process in workflow.onError

### DISCUSSION THREAD

**Discussion by ChenDepp** [nextflow-io/nextflow#5139] [Q&A]
hi @bentsherman 

I want to call process and workflow on workflow.onError and workflow.onComplete, respectively, to make the workflow more flexible. 
how can i do it?

waiting for  your reply, hava a nice day!
thanks

**Comment by bentsherman** [comment]
It is not possible to call a process or workflow from `workflow.onError` or `workflow.onComplete`. You will have to find a way to make it work with regular dataflow logic. It should be possible to do this based on the outputs of upstream processes

**Reply by ChenDepp** [reply]
There are too many independent tasks, and there may be even more tasks in the future. so Dependence on upstream output is too troublesome. can i running shell command by groovy script ?

**Reply by bentsherman** [reply]
If you need to collect outputs from many independent tasks, a topic channel might be a convenient way to do this: https://nextflow.io/docs/latest/channel.html#topic
